settings-portal/Settings.vala

```diff
@@ -0,0 +1,156 @@
+ /*-
+ * Copyright 2021 Alexander Mikhaylenko <alexm@gnome.org>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Library General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU Library General Public
+ * License along with this library; if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor
+ * Boston, MA 02110-1335 USA.
+ */
+
+[DBus (name = "io.elementary.pantheon.AccountsService")]
+private interface Pantheon.AccountsService : Object {
+    public abstract int prefers_color_scheme { owned get; set; }
+}
+
+[DBus (name = "org.freedesktop.Accounts")]
+interface FDO.Accounts : Object {
+    public abstract string find_user_by_name (string username) throws GLib.Error;
+}
+
+private class AccountsServiceMonitor : GLib.Object {
```

Comment:
There are multiple options for this:

- Use Granite instead (and allow Granite to use both portal and accountsservice directly + a way to disable portal there)
- Have this in the same process as the settings daemon itself, then it can be accessed directly from there instead.

Reply:
This one is still an open question. We can also leave it as is, of course.

Reply:
OTOH if it's in the same process Granite will need an override to not try to access portal from portal itself too, having it separate doesn't require it as long as the portal binary doesn't link to granite.